File: tests/journal.lib/chronicler_verbosity.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2021 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// alias
using chronicler_t = pyre::journal::chronicler_t;


// verify that the verbosity level set in the environment matches what the test harness gave me
// on the command line
int main(int argc, char *argv[]) {
    // make sure we get one command line argument
    assert (argc == 2);
    // get the verbosity
    auto verbosity = chronicler_t::verbosity();
    // interpret the command line argument as the intended level
    pyre::journal::verbosity_t expected = std::atoi(argv[1]);
    // verify they match
    assert (verbosity == expected);

    // all done
    return 0;
}


// end of file

File: tests/journal.api/chronicler_verbosity.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2021 all rights reserved


def test():
    """
    Verify that we can control the global verbosity threshold
    """
    # access
    import journal

    # get the global state
    chronicler = journal.chronicler

    # verify that the verbosity level is at its default value
    assert chronicler.verbosity == 1
    # set it to some other value
    chronicler.verbosity = 5
    # verify the assignment sticks
    assert chronicler.verbosity == 5

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
